swim/join_sender.go

```diff
@@ -470,6 +464,23 @@ func (j *joinSender) MakeCall(ctx json.Context, node string, res *joinResponse)
 	return errC
 }
 
+func sendJoinRequest(node *Node, target string, timeout time.Duration) (*joinResponse, error) {
+	ctx, cancel := shared.NewTChannelContext(timeout)
+	defer cancel()
+
+	peer := node.channel.Peers().GetOrAdd(target)
+
+	req := joinRequest{
+		App:         node.app,
+		Source:      node.address,
+		Incarnation: node.Incarnation(),
+		Timeout:     timeout,
+	}
+	res := &joinResponse{}
+	err := json.CallPeer(ctx, peer, node.service, "/protocol/join", req, res)
+	return res, err
+}
+
```

Comment:
I don't like how MakeCall's function signature (returns a channel). This function is a step towards refactoring it away.